loopchain/blockchain/blockchain.py

```diff
@@ -678,7 +685,7 @@ def confirm_prev_block(self, current_block: Block):
             except KeyError:
                 if self.last_block.header.hash == current_block.header.prev_hash:
                     logging.warning(f"Already added block hash({current_block.header.prev_hash.hex()})")
-                    if current_block.header.complained:
+                    if current_block.header.complained and block_manager.epoch.complained_result:
```

Comment:
I think a node can be received valid complained block even if the node have not been received leader complained message.

Reply:
v0.1 block's complained field is not real value. It allow empty block option made same block can treat as a complain block so, It need another checks.